packages/flutter/lib/src/foundation/assertions.dart

```diff
@@ -20,6 +20,96 @@ typedef DiagnosticPropertiesTransformer = Iterable<DiagnosticsNode> Function(Ite
 /// and other callbacks that collect information describing an error.
 typedef InformationCollector = Iterable<DiagnosticsNode> Function();
 
+/// Partial information from a stack frame for stack filtering purposes.
+///
+/// See also:
+///
+///  * [SubStackFilter], which uses this class to compare against [StackFrame]s.
+@immutable
+class PartialStackFrame {
+  /// Creates a new [PartialStackFrame] instance. All arguments are required and
+  /// must not be null.
+  const PartialStackFrame({
+    @required this.className,
+    @required this.method,
+    @required this.packagePath,
+  }) : assert(className != null),
+       assert(method != null),
+       assert(packagePath != null);
+
+  /// The class name for the method.
+  ///
+  /// On web, this is ignored, since class names are not available. Top level
+  /// methods should use the empty string.
```

Comment:
the second sentence here sounds like it relates to the first which confused me at first...

Reply:
Split this up to clarify.